signac/contrib/import_export.py

```diff
@@ -149,6 +149,43 @@ class _SchemaPathEvaluationError(RuntimeError):
     pass
 
 
+def _check_path_function(jobs, path_spec, path_function):
+    """Validate that path function specifies a 1-1 mapping.
+
+    Parameters
+    ----------
+    jobs : iterable of :class:`~signac.contrib.job.Job`
+        A sequence of jobs (instances of :class:`~signac.contrib.job.Job`).
+    path_spec : str
+        The path string that generated the path_function. Displayed in the error message.
+    path_function : callable
+        A callable path generating function.
+
+    Raises
+    -----
+    RuntimeError
+        If paths generated with given path function are not unique.
+
+    """
+    # quick check first
+    links = {path_function(job) for job in jobs}
```

Comment:
One way to avoid re-computing the `path_function` for all jobs if the "quick check" fails would be to use a `Counter`: https://docs.python.org/3/library/collections.html#collections.Counter

Reply:
I added this but didn't investigate the impact on the performance.